components/needs_team/needs_team.tsx

```diff
@@ -202,7 +200,6 @@ export default class NeedsTeam extends React.PureComponent<Props, State> {
         }
         if (Date.now() - this.blurTime > UNREAD_CHECK_TIME_MILLISECONDS && this.props.currentTeamId) {
             this.props.actions.fetchMyChannelsAndMembers(this.props.currentTeamId);
-            this.props.actions.loadProfilesForDirect();
```

Comment:
Despite the name, `loadProfilesForDirect` would update the prefs to ensure unread DMs/GMs were visible in the sidebar when they were unread. Since we always show unread channels in the new sidebar, that's not needed any more.